modular_skyrat/modules/window_airbags/code/window_airbag.dm

```diff
@@ -0,0 +1,101 @@
+/obj/structure/window/reinforced/fulltile/Initialize(mapload, direct)
+	. = ..()
+	AddElement(/datum/element/airbag)
+
+/**
+ * Airbag Element
+ *
+ * Basically a fancy create on destroy. I would love to add a disarm function, but that would mean it would need to be a component.
+ *
+ * And since this is going on every window, an element is more reasonable for performance :)
+ */
+/datum/element/airbag
+	/// The type we spawn when our parent is destroyed
+	var/airbag_type = /obj/item/airbag/immediate_arm
+
+/datum/element/airbag/Attach(datum/target, airbag_type_override)
+	. = ..()
+	if(!ismovable(target))
+		return ELEMENT_INCOMPATIBLE
+
+	RegisterSignal(target, COMSIG_ATOM_DESTRUCTION, .proc/deploy_airbag)
+	RegisterSignal(target, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+
+/datum/element/airbag/Detach(datum/target)
+	UnregisterSignal(target, list(COMSIG_ATOM_DESTRUCTION, COMSIG_PARENT_EXAMINE))
+
+/datum/element/airbag/proc/deploy_airbag(atom/movable/destroying_atom, damage_flag)
+	SIGNAL_HANDLER
+
+	new airbag_type(get_turf(destroying_atom))
+
+/datum/element/airbag/proc/on_examine(datum/source, mob/user, list/examine_text)
+	SIGNAL_HANDLER
+
+	examine_text += span_warning("It has a blinking red light indicating an airbag is primed and ready to trigger on harsh impact.")
+
+
+
+// A fun little gadget!
+/obj/item/airbag
```

Comment:
maybe add the `wires` datum to this to allow it to be hacked and at very least detonated early, if not disarmed?

Reply:
Well, I'll probably make it a wrench to remove tbh